scripts/lib/CIME/case/case.py

```diff
@@ -368,11 +368,11 @@ def get_resolved_value(self, item, recurse=0, allow_unresolved_envvars=False):
 
         return item
 
-    def set_value(self, item, value, subgroup=None, ignore_type=False, allow_undefined=False):
+    def set_value(self, item, value, subgroup=None, ignore_type=False, allow_undefined=False, return_file=False):
         """
         If a file has been defined, and the variable is in the file,
-        then that value will be set in the file object and the file
-        name is returned
+        then that value will be set in the file object and the resovled value
```

Comment:
Please document the return value(s) - i.e., if return_file is False, then just the resolved value is returned; if True, then a tuple (value, filename) is returned. If the variable isn't found, then returns None.